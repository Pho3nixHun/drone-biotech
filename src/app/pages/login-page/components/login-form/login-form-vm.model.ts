export interface LoginFormVM {
    backgroundImageSrc: string;
    descriptionKey: string;
    titleKey: string;
}

export interface LoginFormXVM extends LoginFormVM {
    buttonTitleKey: string;
    emailLabelKey: string;
    passwordLabelKey: string;
    passwordPlaceholderKey: string;
}
