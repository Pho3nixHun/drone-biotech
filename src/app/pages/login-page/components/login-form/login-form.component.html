<div
    class="flex flex-col md:flex-row-reverse h-auto md:h-[75vh] w-full max-w-4xl md:mx-0 bg-white rounded-2xl shadow-md overflow-hidden"
    *transloco="let t"
>
    <div class="md:w-1/2 w-full">
        <img
            [src]="vm().backgroundImageSrc"
            [alt]="t(vm().descriptionKey)"
            class="h-48 md:h-full w-full object-cover"
        />
    </div>
    <div class="flex flex-col justify-center p-5 w-full md:w-1/2 space-y-4">
        <h3 class="text-2xl tracking-wider">
            {{ t(vm().titleKey) }}
        </h3>
        <p>{{ t(vm().descriptionKey) }}</p>
        <p><ng-content /></p>
        <div class="w-full">
            <ng-content select="form" />
        </div>
    </div>
</div>
