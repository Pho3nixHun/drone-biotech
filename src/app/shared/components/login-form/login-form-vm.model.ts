export interface LoginFormVM {
  backgroundImageSrc: string;
}

export interface ExtendedLoginFormVM {
  loginFormVM: LoginFormVM;
  titleKey: string;
  descriptionKey: string;
  buttonTitleKey: string;
}
