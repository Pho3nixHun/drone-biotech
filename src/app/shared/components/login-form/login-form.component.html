<div
    class="flex flex-col md:flex-row-reverse h-auto md:h-[75vh] md:w-[100vh] max-w-4xl mx-5 md:mx-0"
>
    <div
        class="bg-cover bg-no-repeat rounded-2xl h-48 md:h-full md:w-1/2"
        [style.backgroundImage]="'url(' + vm().backgroundImageSrc + ')'"
    ></div>
    <div
        class="flex flex-col items-center justify-center p-5 text-center w-full h-full md:w-1/2"
    >
        <ng-content select="h3, p" />
        <div class="w-full">
            <ng-content select="form" />
        </div>
    </div>
</div>
