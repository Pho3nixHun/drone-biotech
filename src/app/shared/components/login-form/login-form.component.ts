import { Component, input } from '@angular/core';
import { LoginFormVM } from './login-form-vm.model';
/**
 * LoginFormComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering the LoginForm.
 * - Manages the layout and styling of the LoginForm.
 *
 * Out-of-Scope:
 * - Should not handle the data fetching or business logic.
 * - Should not be responsible for manipulating or transforming data.
 *
 * Purpose(optional):
 * - To create a simple layout, a form where the user can login.
 */

@Component({
  selector: 'app-login-form',
  standalone: true,
  imports: [],
  templateUrl: './login-form.component.html',
})
export class LoginFormComponent {
  public vm = input.required<LoginFormVM>();
}
