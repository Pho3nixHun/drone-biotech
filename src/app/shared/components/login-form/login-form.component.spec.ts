import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginFormComponent } from './login-form.component';
import { Component, Input } from '@angular/core';
import { LoginFormVM } from './login-form-vm.model';

@Component({
  template: `<app-login-form [vm]="vm">
    <div>Should not projected</div>
    <p>Should be projected</p>
    <h3>Should be projected</h3>
    <form>Should be projected</form>
  </app-login-form>`,
})
class TestHostComponent {
  @Input() vm!: LoginFormVM;
}

describe('LoginFormComponent', () => {
  let fixture: ComponentFixture<TestHostComponent>;
  let compiled: HTMLElement;
  const vm: LoginFormVM = {
    backgroundImageSrc: 'assets/phoenix.jpg',
  };

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [LoginFormComponent],
      declarations: [TestHostComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(TestHostComponent);
    compiled = fixture.debugElement.nativeElement;
  });

  //Snapshot testing
  it('should project the contents, render the template correctly when there is a VM provided', () => {
    //Arrange
    fixture.componentRef.setInput('vm', vm);
    //Act
    fixture.detectChanges();
    //Assert
    expect(compiled).toMatchSnapshot();
  });
});
