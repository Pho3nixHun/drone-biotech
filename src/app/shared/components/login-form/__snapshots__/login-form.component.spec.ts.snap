// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormComponent should project the contents, render the template correctly when there is a VM provided 1`] = `
<div
  id="root0"
  ng-version="18.2.7"
>
  <app-login-form
    ng-reflect-vm="[object Object]"
  >
    <div
      class="flex flex-col md:flex-row-reverse h-auto md:h-[75vh] md:w-[100vh] max-w-4xl mx-5 md:mx-0"
    >
      <div
        class="bg-cover bg-no-repeat rounded-2xl h-48 md:h-full md:w-1/2"
        style="background-image: url(assets/phoenix.jpg);"
      />
      <div
        class="flex flex-col items-center justify-center p-5 text-center w-full h-full md:w-1/2"
      >
        <p>
          Should be projected
        </p>
        <h3>
          Should be projected
        </h3>
        <div
          class="w-full"
        >
          <form>
            Should be projected
          </form>
        </div>
      </div>
    </div>
  </app-login-form>
</div>
`;
